pkg/controller/linstorcontroller/linstorcontroller_controller.go

```diff
@@ -1054,14 +1054,20 @@ func expectedEndpoint(controllerResource *piraeusv1.LinstorController) string {
 	return lc.DefaultControllerServiceEndpoint(serviceName, useHTTPS)
 }
 
+func getDefaultLabels(controllerResource *piraeusv1.LinstorController) map[string]string {
+	return map[string]string{
+		"app.kubernetes.io/name":       kubeSpec.ControllerRole,
+		"app.kubernetes.io/instance":   controllerResource.Name,
+		"app.kubernetes.io/managed-by": kubeSpec.Name,
+	}
+}
+
 func getObjectMeta(controllerResource *piraeusv1.LinstorController, nameFmt string) metav1.ObjectMeta {
+	defaultLabels := getDefaultLabels(controllerResource)
 	return metav1.ObjectMeta{
-		Name:      fmt.Sprintf(nameFmt, controllerResource.Name),
-		Namespace: controllerResource.Namespace,
-		Labels: map[string]string{
-			"app.kubernetes.io/name":       kubeSpec.ControllerRole,
-			"app.kubernetes.io/instance":   controllerResource.Name,
-			"app.kubernetes.io/managed-by": kubeSpec.Name,
-		},
+		Name:        fmt.Sprintf(nameFmt, controllerResource.Name),
+		Namespace:   controllerResource.Namespace,
+		Labels:      mdutil.MergeStringMap(defaultLabels, controllerResource.ObjectMeta.Labels),
```

Comment:
The order of arguments needs to be reversed here. Otherwise the custom labels overwrite our defaultLabels, and then the selector does not match. This is then rejected by k8s, so the resource can never start.